spanner/jdbc/src/main/java/com/example/spanner/jdbc/JsonInsertDataExample.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.jdbc;
+
+// [START spanner_jdbc_json_insert_data]
+import com.google.cloud.spanner.jdbc.JsonType;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+
+class JsonInsertDataExample {
+  // Class to contain Venue sample data.
+  static class Venue {
+    final long venueId;
+    final String venueDetails;
+
+    Venue(long venueId, String venueDetails) {
+      this.venueId = venueId;
+      this.venueDetails = venueDetails;
+    }
+  }
+
+  static final List<Venue> VENUES =
+      Arrays.asList(
+          new Venue(
+              4,
+              "[{\"name\":\"room 1\",\"open\":\"true\"},"
+                  + "{\"name\":\"room 2\",\"open\":\"false\"}]"),
+          new Venue(19, "{\"rating\":\"9\",\"open\":\"true\"}"),
+          new Venue(
+              42,
+              "{\"name\":null,"
+                  + "\"open\":{\"Monday\":\"true\",\"Tuesday\":\"false\"},"
+                  + "\"tags\":[\"large\",\"airy\"]}"));
+
+  static void insertJsonData() throws SQLException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    insertJsonData(projectId, instanceId, databaseId);
+  }
+
+  static void insertJsonData(String projectId, String instanceId, String databaseId)
+      throws SQLException {
+    String connectionUrl =
+        String.format(
+            "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s",
+            projectId, instanceId, databaseId);
+    try (Connection connection = DriverManager.getConnection(connectionUrl)) {
+      try (PreparedStatement ps =
+          connection.prepareStatement("INSERT INTO Venues\n"
+                      + "(VenueId, VenueDetails)\n"
+                      + "VALUES\n"
+                      + "(?, ?)")) {
+        for (Venue venue : VENUES) {
+          ps.setLong(1, venue.venueId);
+          // Tell the JDBC driver that we want to set a JSON value and not a STRING value
+          // by specifying the JsonType SQL type.
+          ps.setObject(2, venue.venueDetails, JsonType.INSTANCE);
```

Comment:
Do you think this is a more natural way to specify the type other than `ps.setObject(2, Value.json(venue.venueDetails))`?

Reply:
Not necessarily, and I was a little bit in doubt here what is the best to show:
- `ps.setObject(2, Value.json(venue.venueDetails))` is the easiest and prettiest, and would be what I would recommend that users that have compile time access to the Spanner JDBC driver should use.
- Users who do not have access to the JDBC driver on compile time (that is; the application only works with the generic JDBC interfaces and the driver is loaded dynamically by the environment) must use one of the other options.

Maybe we should create two different samples for this to show both. That is what we decided to do for creating a retry loop for aborted transactions:
- JDBC only: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/spanner/jdbc/src/main/java/com/example/spanner/jdbc/TransactionWithRetryLoopUsingOnlyJdbcExample.java
- With compile time access to the JDBC driver: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/spanner/jdbc/src/main/java/com/example/spanner/jdbc/TransactionWithRetryLoopExample.java